### ISSUE TITLE
About rec model output shape?

### ISSUE THREAD

**Issue by lloydnguyen96** [PaddlePaddle/PaddleOCR#2149]
For model: non transformation + resnet34 + bilstm + ctc (linear) output with shape (N=batch_size, T=time_steps, C=num_classes)
If the model with the above architecture is trained for Chinese, then what are the values used for N, T and C in this repo? Thank you!

**Comment by littletomatodonkey** [comment]
```
N=any value you seet
T=80(if the input shape is 32x320, then the feature map shape after the backbone is 1x80)
C=6625(word num in the dict)
```

**Comment by lloydnguyen96** [comment]
Did you train a model on this repo with this configuration (same architecture + Chinese)? Is T=80 pretty large for a Chinese text line (I see they chose 25 for max_text_length)? is it feasible to train using CTC loss for such a big output (BS, 80, 6625)? I faced problem training with output (BS, 65, 2002) using CTCLoss in Pytorch several weeks already

**Comment by lloydnguyen96** [comment]
When I reduce the number_of_classes -> Loss converges but still slower than expected.

**Comment by littletomatodonkey** [comment]
1. Actually we find that if you reduce T (the feature map resolution reduces at the same time), the model accuracy may has a slightly drop.
2. When the num_class is very large, it is very difficult to converge, that's why we add so much data for the training process.

**Comment by lloydnguyen96** [comment]
Is an unbalanced dataset a big impact on the training process?
Here is an example:
Character          Number of occurences
"1": 12169,
    "0": 11490,
    "8": 9113,
    "3": 7405,
    "5": 7227,
    "2": 6822,
    "6": 6760,
    "7": 6262,
    "9": 6156,
    "4": 4992,
    "A": 4794,
...
"\u6c35": 1,
    "\u6c65": 1,
    "\u6ba1": 1,
    "\u846c": 1,
    "\u9773": 1,
    "\u9bf0": 1,
    "\u63c9": 1,
    "\u95ed": 1,
    "\u775b": 1,
    "\u54d1": 1,
    "\u4f5a": 1

**Comment by lloydnguyen96** [comment]
C=6625=6623 character classes (in PaddleOCR/ppocr/utils/ppocr_keys_v1.txt) + 1 (for CTC blank) + 1 (other characters), right?
